server/src/service/item.rs

```diff
@@ -0,0 +1,124 @@
+// org.freedesktop.Secret.Item
+
+use std::collections::HashMap;
+
+use oo7::dbus::api::properties::Properties;
+use serde::{Deserialize, Serialize};
+use zbus::{dbus_interface, fdo, zvariant, ObjectServer};
+use zvariant::{ObjectPath, OwnedObjectPath};
+
+use crate::KEYRING;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Item {
+    locked: bool,
+    attributes: Properties,
+    created: u64,
+    modified: u64,
+    secret: Vec<u8>,
+    path: OwnedObjectPath,
+}
+
+#[dbus_interface(name = "org.freedesktop.Secret.Item")]
+impl Item {
+    pub async fn delete(
+        &self,
+        #[zbus(object_server)] object_server: &ObjectServer,
+    ) -> fdo::Result<ObjectPath> {
+        let mut properties: HashMap<&str, &str> = Default::default();
+
+        for (key, value) in self.attributes().await.attributes().unwrap().into_iter() {
+            properties.insert(key.as_str(), value.as_str());
+        }
+
+        let _ = KEYRING.get().unwrap().delete(&properties).await;
+        let _ = object_server.remove::<Item, _>(self.path().await).await;
+        Ok(ObjectPath::default().into())
+    }
+
+    pub async fn get_secret(&self /*session: Session*/) -> fdo::Result<Vec<u8>> {
```

Comment:
The session should be an object path, as creating one would require some extra-logic that you can't do in the Deserialize implementation.